src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy

```diff
@@ -514,106 +496,137 @@ class OpenShiftDSL implements Serializable {
     /**
      * The select operation can be executed multiple ways:
      *      selector()   // Selects all
-     *      selector( "pod" )  // Selects all of a name
-     *      selector( "dc", "jenkins" )   // selects a particular instance dc/jenkins
-     *      selector( "dc/jenkins" )   // selects a particular instance dc/jenkins
-     *      selector( "dc", [ alabel: 'avalue' ] ) // Selects using label values
+     *      selector("pod")  // Selects all of a name
+     *      selector("dc", "jenkins")   // selects a particular instance dc/jenkins
+     *      selector("dc/jenkins")   // selects a particular instance dc/jenkins
+     *      selector("dc", [ alabel: 'avalue' ]) // Selects using label values
      * When labels are used, the qualifier will be a map. In other cases, expect a String or null.
      */
-    public OpenShiftResourceSelector selector(Object kind = null, Object qualifier=null ) {
-        return new OpenShiftResourceSelector( "selector", kind, qualifier );
+    public OpenShiftResourceSelector selector(Object kind = null, Object qualifier=null) {
+        return new OpenShiftResourceSelector("selector", kind, qualifier);
     }
 
     private OpenShiftResourceSelector objectDefAction(String verb, Object obj, Object[] userArgs) {
-        obj = toSingleObject( obj ); // Combine a list of objects into a single OpenShift List model, if necessary.
+        obj = toSingleObject(obj); // Combine a list of objects into a single OpenShift List model, if necessary.
 
-        if ( obj instanceof Map ) {
-            obj = JsonOutput.toJson( obj );
+        if (obj instanceof Map) {
+            obj = JsonOutput.toJson(obj);
         }
 
         String s = obj.toString();
         boolean markup = s.contains("{") || s.contains(":"); // does this look like json or yaml?
         boolean httpref = s.toLowerCase().startsWith("http") && verb.equalsIgnoreCase("create"); // a create from a http or https raw.githubuser path
 
-        Result r = new Result( verb );
+        Result r = new Result(verb);
 
-        if ( markup && !httpref) {
-            FilePath f = currentContext.exec.getWorkspaceFilePath().createTextTempFile( verb, ".markup", s, false );
+        if (markup) {
```

Comment:
so an http url will always have a `:` in it, so `markup` will always be true if `httpref` is true

which means we'll never hit the `else if (httpref)` block

seems like this needs more refactoring